.github/workflows/plonk-artifacts.yml

```diff
@@ -0,0 +1,74 @@
+# Generates and uploads new Plonk artifacts to S3 when  and manual release
+# On `push`, the version/file name is the commit hash
+# Use `workflow_dispatch` for Sphinx releases, with manually specified version name
+name: Update Plonk artifacts in S3
+
+on:
+  push:
+    branches: dev
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Artifact version used for archive name, e.g. v1.0.0-testnet"
+        type: string
+        required: true
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  update-plonk-artifacts:
+    name: Update Plonk artifacts
+    runs-on: warp-ubuntu-latest-x64-32x
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: argumentcomputer/ci-workflows
+      - uses: ./.github/actions/ci-env
+      - uses: actions/checkout@v4
+      - name: Setup CI
+        uses: ./.github/actions/setup
+        with:
+          pull_token: ${{ secrets.REPO_TOKEN }}
+      - name: Install AWS CLI
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y awscli
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          aws-access-key-id: ${{ secrets.S3_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.S3_SECRET_KEY }}
+          aws-region: us-east-2
+      - name: Check for existing artifacts
+        id: check-s3
+        run: |
+          if [[ "${{ github.event_name }}" == "push" ]]; then
+            VERSION=$(grep -r "const SPHINX_CIRCUIT_VERSION" core/src/lib.rs | awk -F '"' '{print $2}')
+          else
+            VERSION=${{ inputs.version }}
+          fi
+
+          BUCKET_NAME="sphinx-plonk-params"
+
+          FILE=$(aws s3 ls "s3://${BUCKET_NAME}/${VERSION}.tar.gz")
+
+          if [[ -n "$FILE" ]]; then
+            echo "File ${VERSION}.tar.gz exists in $BUCKET_NAME"
+            echo "needs-update=false" | tee -a $GITHUB_OUTPUT
+          else
+            echo "needs-update=true" | tee -a $GITHUB_OUTPUT
+          fi
```

Comment:
So, with this logic, I could be passing a manual VERSION, but I cannot use that manual workflow dispatch to overwrite the existing bucket, which is confusing me:
- why is VERSION required on workflow_dispatch? can't we make it default to SPHINX_CIRCUIT_VERSION?
- is there a use case for having a manual VERSION that would perhaps differ from the code SPHINX_CIRCUIT_VERSION, considering the matching code dictates through SPHINX_CIRCUIT_VERSION which circuit parameters will be uploaded? I am not sure, but perhaps. 
- where I can see this definitely being useful is to have a workflow_dispatch force the re-creation of the parameters irrespective of whether the bucket exists and is populated (e.g. in case the parameters are corrupt), could we have that?

Reply:
> why is VERSION required on workflow_dispatch? can't we make it default to SPHINX_CIRCUIT_VERSION?

Yes that makes sense to have a default.

> is there a use case for having a manual VERSION that would perhaps differ from the code SPHINX_CIRCUIT_VERSION, considering the matching code dictates through SPHINX_CIRCUIT_VERSION which circuit parameters will be uploaded? 

Perhaps if you want to upload some development parameters that aren't a fully versioned release? I was following the logic of https://github.com/argumentcomputer/sphinx/blob/dev/prover/release.sh, which allows you to specify an arbitrary string and refers to naming by commit hash in the comments. cc @wwared 

> where I can see this definitely being useful is to have a workflow_dispatch force the re-creation of the parameters irrespective of whether the bucket exists and is populated (e.g. in case the parameters are corrupt), could we have that?

Definitely could use it to force update instead. Where would we upload the parameters if there's no bucket though?

Reply:
So, I'm agnostic w.r.t the existence of the VERSION parameter, and happy if it exists, as long as it has a decent default (aka, if `SPHINX_CIRCUIT_VERSION` is defined as a bash variable with the value found in the local commit's file: `VERSION=${VERSION:-$SPHINX_CIRCUIT_VERSION}`).

Update forcing using the `workflow_dispatch` flow with an explicit version can simply create a bucket at the file emplacement suggested by the value of `VERSION` if there is none.